File: problems/1812-determine-color-of-a-chessboard-square/solution.java
class Solution {
    public boolean squareIsWhite(String coordinates) {
        char file = coordinates.charAt(0);
        char rank = coordinates.charAt(1);

        if ((file - 'a') % 2 == 0) {
            return (rank - '1') % 2 != 0;
        } else {
            return (rank - '1') % 2 == 0;
        }
    }
}

File: problems/1812-determine-color-of-a-chessboard-square/solution.c
bool squareIsWhite(char* coordinates) {
    int column = coordinates[0] - 'a' + 1;
    int row = coordinates[1] - '0';

    if ((column + row) % 2 == 0) {
        return false;
    } else {
        return true;
    }
}